source/special.tex

```diff
@@ -3203,17 +3203,23 @@
 \end{example}
 
 \pnum
-When the criteria for elision of a copy/move operation are met,
-but not for an \nonterminal{exception-declaration},
-and the object
-to be copied is designated by an lvalue,
-or when the \grammarterm{expression} in a \tcode{return} statement
+In the following copy-initialization contexts, a move constructor might be used instead of a copy constructor:
+\begin{itemize}
+\item If the \grammarterm{expression} in a \tcode{return} statement~(\ref{stmt.return})
 is a (possibly parenthesized) \grammarterm{id-expression}
 that names an object with automatic storage duration declared in the body
 or \grammarterm{parameter-declaration-clause} of the innermost enclosing
-function or \grammarterm{lambda-expression},
+function or \grammarterm{lambda-expression}, or
+
+\item if the operand of a \grammarterm{throw-expression}~(\ref{expr.throw})
+is the name of a non-volatile automatic object
+(other than a function or catch-clause parameter)
+whose scope does not extend beyond the end of the innermost enclosing
+\grammarterm{try-block} (if there is one),
```

Comment:
Err, right, bad example, a correct example would be more like this:
```
void f() {
  exception_thrown_on_failure f;
  try { /*...*/ }
  catch (...) {
    throw f; // no move here =(
  }
}
```
I see no reason we couldn't use a move here. Anyway, yes, I'll take this to the core reflector.